src/util/propertyTreeHelpers.ts

```diff
@@ -1,7 +1,125 @@
-import { PropertyOwners } from '@/types/types';
+import {
+  Properties,
+  PropertyDetails,
+  PropertyOwner,
+  PropertyOwners,
+  PropertyVisibility
+} from '@/types/types';
 
 import { InterestingTagKey } from './keys';
 
-export function hasInterestingTag(propertyOwners: PropertyOwners, uri: string) {
+export function identifierFromUri(uri: string) {
+  // The identifier is always the last word in the URI
+  const identifier = uri.split('.').pop();
+  if (!identifier) {
+    throw Error(`Tried to get identifier from invalid URI '${uri}'`);
+  }
+  return identifier;
+}
+
+export function sgnIdentifierFromSubownerUri(uri: string) {
+  const splitUri = uri.split('.');
+  if (splitUri.length < 2 || splitUri[0] !== 'Scene') {
+    throw Error(`Invalid SGN URI '${uri}'`);
+  }
+  return splitUri[1];
+}
+
+export function displayName(propertyOwner: PropertyOwner) {
+  return propertyOwner.name ?? propertyOwner.identifier ?? propertyOwner.uri;
+}
+
+export function hasInterestingTag(uri: string, propertyOwners: PropertyOwners) {
   return propertyOwners[uri]?.tags.some((tag) => tag.includes(InterestingTagKey));
 }
+
+export function guiOrderingNumber(
+  properties: Properties,
+  uri: string
+): number | undefined {
+  const shouldUseGuiOrderingNumber = properties[`${uri}.UseGuiOrdering`]?.value || false;
+  if (!shouldUseGuiOrderingNumber) {
+    return undefined;
+  }
+  return properties[`${uri}.GuiOrderingNumber`]?.value as number | undefined;
+}
+
+export function isRenderable(uri: string) {
+  const renderableSuffix = '.Renderable';
+  return uri.endsWith(renderableSuffix);
+}
+
+export function isGlobe(renderableUri: string, properties: Properties) {
+  const renderableSuffix = '.Renderable';
+  return (
+    renderableUri.endsWith(renderableSuffix) &&
+    properties[`${renderableUri}.Type`]?.value === 'RenderableGlobe'
+  );
+}
+
+export function isGlobeLayersUri(uri: string, properties: Properties) {
+  const suffix = '.Renderable.Layers';
+  return uri.endsWith(suffix) && isGlobe(uri.replace('.Layers', ''), properties);
+}
+
+export function isPropertyOwnerHidden(properties: Properties, uri: string) {
+  const prop = properties[`${uri}.GuiHidden`];
+  if (prop && prop.value) {
+    return true;
+  }
+  return false;
```

Comment:
The property value `prop.value` should always be a boolean, so this is not a problem here, but I can see how it feels unclear. 

This is a direct rewrite of the old code. A potential alternative way to write it could be: 
```
const isHidden = properties[`${uri}.GuiHidden`]?.value as boolean | undefined;
return isHidden || false;
```

Reply:
Ah okay makes sense, I think the alternative is clearer. But I'm not a big fan of us having to explicitly type `as` everywhere since that can lead to bugs down the road. I think at some point we should revisit the `property.value` and see if we can come up with a better solution that does not require us to type cast it whenever we need to use it.

Reply:
I agree with that - i.e. to avoid using "as". 

However, I also feel like this typing business maybe not have to be perfect, as long as it works and is understnadable. To some degree, we also need to balance effort with benefit.

Reply:
But I'll change it according to the suggested alternative